### PR TITLE
SDK and CLI code samples for AML Spark Public Preview

### PR DESCRIPTION
**Pull request by ynpandey**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: cli/jobs/spark/user-assigned-identity.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "properties":{
+    },
+    "location": "<AZURE_REGION>",
```
Comment (harneetvirk) [inline cli/jobs/spark/user-assigned-identity.json:4]: How are the values of these templates for e.g. <AZURE_REGION> getting populated?

File: sdk/python/jobs/spark/attached_spark_pipeline_default_identity.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+@dsl.pipeline(description="Sample Pipeline with Spark component")
+def spark_pipeline(spark_input_data):
+    spark_step = spark_component(titanic_data=spark_input_data)
+    spark_step.inputs.titanic_data.mode = InputOutputModes.DIRECT
+    spark_step.outputs.wrangled_data = Output(
+        type="uri_folder",
+        path="azureml://datastores/workspaceblobstore/paths/data/wrangled/",
+    )
+    spark_step.outputs.wrangled_data.mode = InputOutputModes.DIRECT
+    spark_step.compute = "<ATTACHED_SPARK_POOL_NAME>"
```
Comment (harneetvirk) [inline sdk/python/jobs/spark/attached_spark_pipeline_default_identity.py:41]: How is the value of <ATTACHED_SPARK_POOL_NAME> getting updated in automated workflows?

**Comment by harneetvirk** [comment]
No workflows have been added for these newly added resources or scripts. In case, someone tries to run python readme.py files to generate the workflows, that person will end up with all these new workflows and be blocker for his changes.
FYI: @balapv

**Review by balapv — COMMENTED** [review]
1. We have used notebook files for all other examples in python/sdk. Can we convert your samples also into notebooks instead of python scripts. That way you can add some minimal documentation as well
1. Where are we ensuring that the data required for these jobs (both CLI and SDK) are available? Do we have a script to setup the required datastore/dataset? If this sort of setup is not available, whenever we move samples testing to a new workspace, they will fail

File: cli/jobs/spark/user-assigned-identity.json (hunk 1 above)
Comment (balapv) [reply cli/jobs/spark/user-assigned-identity.json:4]: @harneetvirk there are placeholders in the samples like <STORAGE_ACCOUNT_NAME>, <BLOB_CONTAINER_NAME> and <FILE_SYSTEM_NAME> which user has to replace before running them. 
Do we have a recommendation for this scenario? How will we automate testing for these? What do you think?

Comment (ynpandey) [reply cli/jobs/spark/user-assigned-identity.json:4]: As discussed, the samples will be excluded from automation run.

File: sdk/python/jobs/spark/attached_spark_pipeline_default_identity.py (hunk 2 above)
Comment (ynpandey) [reply sdk/python/jobs/spark/attached_spark_pipeline_default_identity.py]: As discussed, the samples will be excluded from automation run.

**Review by harneetvirk — APPROVED** [review]

**Pull request merged**